EIPS/eip-draft_Precompiled+contracts+for+ring+signature+verification+on+the+elliptic+curve+alt_bn128__abbrev.md

```diff
@@ -0,0 +1,72 @@
+## Preamble
+
+    EIP: 
+    Title: Precompiled contracts for ring signature verification on the elliptic curve alt_bn128.
+    Author: Jack Lu <jack@wanchain.org>, Demmon Guo <demmon@wanchain.org> and Shi Liu <shi@wanchain.org>
+    Type: Standard Track
+    Category (*only required for Standard Track): Core 
+    Status: Draft
+    Created: 2017-08-21
+    
+
+
+## Simple Summary
+Use precompiled contracts for ring signature verification.
+
+## Abstract
+This EIP suggests adding precompiled contracts for ring signature verification on alt_bn128. This can in turn be combined with EIP #196 to verify ring signature in Ethereum smart contracts. The general benefit of this implementation is that it will offer another solution for privacy protection on Ethereum.
+
+## Motivation
+Currently, ether transactions on Ethereum are transparent, which makes them unsuitable for some use-cases where users’ privacy is of concern. EIP #196 and EIP #197 suggest adding precompiled contracts for addition and scalar multiplication on a specific pairing-friendly elliptic curve to verify zkSNARKs in Ethereum smart contract. But that implemtation has high computation complexity. Monero solves this problem by using Cryptonote under UTXO model. However, this technology cannot be directly used in Ethereum under account model. Considering all of the above, this EIP forgoes zero knowledge proof and proposes to implement precompiled contracts for ring signature verification.
+
+## Specification
+**Common parameters:**  
+
+*q*:  a prime number;  
+*E*: an elliptic curve equation;  
+*G*: a base point;  
+*Hs*: a cryptographic hash function which maps string of {0,1} to element of *F_q*;  
+*Hp*: a hash function which creates a mapping between elements of *E(F_q)*;  
++: point addition on elliptic curve;  
+*: scalar multiplication on elliptic curve.  
+
+**Ring signature verification**  
+
+Signature σ consists of a set of public keys, two arrays of random numbers selected by signer and a point I called key image.   

```

Comment:
The Franklin, Zhang signature algorithm and its relation to coin tumbling is described in the following IACR paper: https://eprint.iacr.org/2017/881.pdf

Implementations in Solidity and Go for the Ethereum alt_bn128 curve are located at https://github.com/clearmatics/mobius and https://github.com/clearmatics/orbital